js/libs/keycloak-js/src/keycloak.js

```diff
@@ -1317,9 +1329,39 @@ function Keycloak (config) {
                     return createPromise().promise;
                 },
 
-                logout: function(options) {
-                    window.location.replace(kc.createLogoutUrl(options));
-                    return createPromise().promise;
+                logout: async function(options) {
+
+                    const logoutMethod = options?.logoutMethod ?? kc.logoutMethod;
+                    if (logoutMethod === "GET") {
+                        window.location.replace(kc.createLogoutUrl(options));
+                        return createPromise().promise;
+                    }
+
+                    const logoutUrl = kc.createLogoutUrl(options);
+                    const logoutRequest = {
+                        method: "POST",
+                        headers: {
+                            "Content-Type": "application/x-www-form-urlencoded"
+                        },
+                        body: new URLSearchParams({
+                            id_token_hint: kc.idToken,
+                            client_id: kc.clientId,
+                            post_logout_redirect_uri: adapter.redirectUri(options, false)
+                        })
+                    };
+
+                    const response = await fetch(logoutUrl, logoutRequest);
+                    if (response.redirected) {
+                        window.location.href = response.url;
+                        return;
+                    }
+
+                    if (response.ok) {
+                        window.location.reload();
+                        return;
+                    }
+
+                    throw "Logout failed";
```

Comment:
Let's wrap this in an `Error` constructor, and perhaps make the message a bit more explicit.
```suggestion
                    throw new Error("Logout failed, request returned an error code.");
```